### PR TITLE
IRO-1663 - Introduced styled buttons.

### PR DESCRIPTION
**Pull request by azhurauski-sc**
Introduced styled buttons.

Made two types of button `primary` and `secondary` with different size options.
For link there will be `text_link`, it's not clear to me how icon in link should be colorized as it has multiple combinations so make it the same color as text in link.
For Button Group used chakra Menu without `as={IconButton}` option as it seems in that case it need some additional specification for button or IconButton, instead added styling for icon nested inside MenuButton component.

Also there was two color that doesn't have constants, don't have good name suggestions for them
```
  DARKER_GREY_1: '#222222',
  DARKER_GREY_2: '#626262',
 ```

<details> 
<summary>Demo </summary>

demo1

Disabled buttons

demo2

</details>

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -28,6 +31,11 @@ module.exports = {
       },
       resolve: {
         ...resolve,
+        alias: {
+          ...resolve.alias,
+          "@emotion/core": toPath("node_modules/@emotion/react"),
+          "emotion-theming": toPath("node_modules/@emotion/react"),
+        },
```
Comment (azhurauski-sc) [inline .storybook/main.js:38]: this part is required to make chakra-ui theme work properly in storybook.

File: .storybook/preview.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import React from 'react';
```
Comment (ndemidovich-sc) [inline .storybook/preview.js:1]: In React 17 no need to have React import.

File: stories/Button.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import React from 'react'
```
Comment (ndemidovich-sc) [inline stories/Button.stories.tsx:1]: In React 17 no need to have React import.

File: theme/buttonsTheme.ts [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+const buttonsTheme: ComponentStyleConfig = {
+  baseStyle: {
+    fontStyle: 'normal',
+    fontWeight: '400',
+    fontFamily: 'ABC Favorit Trial',
+    lineHeight: '160%',
+  },
+  sizes: {
+    large: {
+      h: '69px',
```
Comment (ndemidovich-sc) [inline theme/buttonsTheme.ts:14]: try to use `rem` instead of `px`

**Review by ndemidovich-sc — CHANGES_REQUESTED** [review]
Also if you create customization for components as separated files, then I suggest to create folder `components`, put customization in it and create index file with merged customization.

File: .storybook/preview.js (hunk 2 above)
Comment (ndemidovich-sc) [reply .storybook/preview.js:1]: > In React 17 no need to have React import.

Not fixed

File: theme/components/buttonsTheme.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
+import type { ComponentStyleConfig } from '@chakra-ui/theme'
+import { mode } from '@chakra-ui/theme-tools'
+import { NAMED_COLORS } from 'styles/constants'
```
Comment (ndemidovich-sc) [inline theme/components/buttonsTheme.ts:3]: Merge last changes from `master` and use `theme/constants` instead of `styles/constants`

**Review by ndemidovich-sc — CHANGES_REQUESTED** [review]

File: theme/components/buttonsTheme.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      },
+      ':hover': {
+        color: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+        '.chakra-button__icon': {
+          fill: mode(NAMED_COLORS.DEEP_BLUE, NAMED_COLORS.WHITE)(props),
+        },
+      },
+    }),
+  },
+}
```
Comment (brekk) [inline theme/components/buttonsTheme.ts:115]: Probably fine as is, but it looks like there might be a reusable set of definitions we could have to keep things more DRY ?

**Review by brekk — APPROVED** [review]
Generally good as is, I mentioned a file which maybe could be simplified, but perhaps that would be pre-emptive.

File: theme/components/buttonsTheme.ts (hunk 6 above)
Comment (ndemidovich-sc) [reply theme/components/buttonsTheme.ts:115]: For now it is ok. We need to create more components customization and only after that move some code in common parts.

**Review by ndemidovich-sc — APPROVED** [review]

**Comment by ndemidovich-sc** [comment]
@azhurauski-sc Resolve conflicts and merge it. (Don't forget about squash)

**Pull request merged**